host/sgx/create.c

```diff
@@ -895,7 +924,8 @@ oe_result_t oe_sgx_build_enclave(
         tls_page_count,
         &props,
         &loaded_enclave_pages_size,
-        &enclave_size));
+        &enclave_size,
```

Comment:
The `enclave_size` will be set to the enclave image
https://github.com/openenclave/openenclave/blob/a5f59b2d6048cf71128c540e473069bf06fb31c8/host/sgx/create.c#L951

https://github.com/openenclave/openenclave/blob/a5f59b2d6048cf71128c540e473069bf06fb31c8/host/sgx/loadelf.c#L922

And the enclave will use the value to determine the end of the enclave memory range

https://github.com/openenclave/openenclave/blob/a5f59b2d6048cf71128c540e473069bf06fb31c8/enclave/core/sgx/memory.c#L12

Now for the 0-base enclave, the `enclave_size` is normal enclave_size + start_address, which will mess up the enclave memory range check. Can you double check?

Reply:
Fixed the bug introduced in address within enclave check. Check for other occurances too.